File: CodeSlice.UnitTesting/CodeSlice.UnitTesting.NUnit/VoteTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using System.ComponentModel.DataAnnotations;
using System.Collections;

namespace CodeSlice.UnitTesting.NUnit
{
    [TestFixture]
    public class VoteTest : ValidationTest
    {
        [Test]
        public void VoteWithValidValueTest1()
        {
            Model.Vote vote = new Model.Vote
            {
                Value = -1
            };

            ICollection errors = Validate(vote, "Value");
            Assert.IsEmpty(errors);
        }

        [Test]
        public void VoteWithValidValueTest1()
        {
            Model.Vote vote = new Model.Vote
            {
                Value = 1
            };

            ICollection errors = Validate(vote, "Value");
            Assert.IsEmpty(errors);
        }
    }
}

File: CodeSlice.UnitTesting/CodeSlice.UnitTesting.Web/qunit/unit/Vote.test.js
module('Vote Model Validation');

test('Invalid Vote Value (-2)', function(){
    var vote = new Vote({ value: -2 });
    equals(vote.validate().length, 1);
});

test('Invalid Vote Value (0)', function(){
    var vote = new Vote({ value: 0 });
    equals(vote.validate().length, 1);
});

test('Invalid Vote Value (2)', function(){
    var vote = new Vote({ value: 2 });
    equals(vote.validate().length, 1);
});

test('Valid Vote Value (-1)', function(){
    var vote = new Vote({ value: -1 });
    equals(vote.validate().length, 0);
});

test('Valid Vote Value (1)', function(){
    var vote = new Vote({ value: 1 });
    equals(vote.validate().length, 0);
});